src/prefect/runner/storage.py

```diff
@@ -295,17 +360,25 @@ async def _clone_repo(self):
         # Add the clone command and its parameters
         cmd += ["clone", repository_url]
 
-        if self._branch:
-            cmd += ["--branch", self._branch]
         if self._include_submodules:
             cmd += ["--recurse-submodules"]
 
         # This will only checkout the top-level directory
         if self._directories:
             cmd += ["--sparse"]
 
-        # Limit git history and set path to clone to
-        cmd += ["--depth", "1", str(self.destination)]
+        if self._commit_sha:
+            cmd += ["--filter=blob:none"]
```

Comment:
If we're going to clone this way, we'll need to fetch the contents of the commits to ensure we get the actual files. Something like `git fetch origin COMMIT_SHA`. It looks like this flag is often used with `--no-checkout` too to avoid pulling files for HEAD.

Overall, I think we want the initial clone sequence to look something like this:
```base
git clone --filter=blob:none --no-checkout REPO_URL DESTINATION
git fetch origin COMMIT_SHA # runs in DESTINATION
git checkout COMMIT_SHA # runs in DESTINATION
```